catroid/src/main/java/org/catrobat/catroid/ui/fragment/ProjectListFragment.java

```diff
@@ -369,24 +372,28 @@ public boolean onOptionsItemSelected(MenuItem item) {
 	}
 
 	private void initializeDefaultProjectAfterDelete() {
-		final ProjectManager projectManager = ProjectManager.getInstance();
-		getActivity().findViewById(R.id.fragment_container).setVisibility(View.GONE);
-		getActivity().findViewById(R.id.progress_circle).setVisibility(View.VISIBLE);
-		Runnable r = new Runnable() {
+		final Activity activity = getActivity();
+		projectInitializer = new AsyncTask<Void, Void, Activity>() {
+
+			@Override
+			protected void onPreExecute() {
+				activity.findViewById(R.id.fragment_container).setVisibility(View.GONE);
+				activity.findViewById(R.id.progress_circle).setVisibility(View.VISIBLE);
+			}
+
+			@Override
+			protected Activity doInBackground(Void... params) {
+				ProjectManager.getInstance().initializeDefaultProject(activity);
+				return activity;
```

Comment:
IMO there is no value in returning activity.
The return value of `doInBackground` is the result of the Task. Since the task has no result (just side-effects), it's fine to return `null`. Don't forget to update `onPostExecute`.